staging/src/k8s.io/apiserver/pkg/authentication/request/x509/x509.go

```diff
@@ -276,10 +277,17 @@ var CommonNameUserConversion = UserConversionFunc(func(chain []*x509.Certificate
 	if len(chain[0].Subject.CommonName) == 0 {
 		return nil, false, nil
 	}
+
+	fp := sha256.Sum256(chain[0].Raw)
+	id := "X509SHA256=" + hex.EncodeToString(fp[:])
+
 	return &authenticator.Response{
 		User: &user.DefaultInfo{
 			Name:   chain[0].Subject.CommonName,
 			Groups: chain[0].Subject.Organization,
+			Extra: map[string][]string{
+				user.CredentialIDKey: {id},
```

Comment:
Drive by comment, I would also put the SubjectKeyID in here: https://datatracker.ietf.org/doc/html/rfc5280#section-4.2.1.2

While there's no guarantee that it's a unique ID, it is useful to cross reference against logs e.g. certificate authority issuance logs.

Reply:
Many CAs use the sha256 sum of the public key for SKID so they will often be the same. I'd say it's fine to start with what we have here.